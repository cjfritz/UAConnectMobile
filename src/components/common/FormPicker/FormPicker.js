import React from 'react';
import {
  Picker, Icon, Header, Left, Button, Text, Body, Right, Item,
} from 'native-base';
import styles from './FormPicker.style';

const FormPicker = ({
  placeholder,
  selectedValue,
  onValueChange,
  items,
}) => (
  <Item picker>
    <Picker
      iosIcon={ <Icon name='ios-arrow-down' color='darkred' style={ styles.icon } /> }
      placeholder={ placeholder }
      selectedValue={ selectedValue() }
      onValueChange={ value => onValueChange(value) }
      renderHeader={ backPress => (
        <Header style={ styles.pickerHeader }>
          <Left style={ { flex: 1 } }>
            <Button transparent onPress={ backPress } style={ { flexDirection: 'row' } }>
              <Icon name='ios-arrow-down' color='white' />
              <Text style={ { fontSize: 15 } }>Close</Text>
            </Button>
          </Left>
          <Body style={ { flex: 1 } }>
            <Text numberOfLines={ 1 } style={ styles.titleText } adjustsFontSizeToFit>
              { placeholder }
            </Text>
          </Body>
          <Right style={ { flex: 1 } } />
        </Header>
      ) }
    >
      {
      items.map((value, index) => (
        <Picker.Item
          label={ value }
          value={ value }
          key={ index } // eslint-disable-line react/no-array-index-key
        />
      ))
    }
    </Picker>
  </Item>
);

export default FormPicker;
