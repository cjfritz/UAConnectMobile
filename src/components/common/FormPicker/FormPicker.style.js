import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  icon: {
    width: '100%',
  },
  pickerHeader: {
    backgroundColor: 'darkred',
    flexDirection: 'row',
  },
  titleText: {
    color: 'white',
    fontSize: 20,
  },
});
