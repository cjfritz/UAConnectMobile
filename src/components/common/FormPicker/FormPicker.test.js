import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import FormPicker from './FormPicker';

describe('<FormPicker />', () => {
  it('can render with initial state', () => {
    const props = {
      placeholder: 'testPlaceholder',
      // selectedValue: jest.fn().mockImplementation(() => 'value'),
      // selectedValue: jest.fn(() => 'item1'),
      selectedValue: jest.fn(),
      onValueChange: jest.fn(),
      items: [
        'item1',
        'item2',
        'item3',
      ],
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<FormPicker { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('can render selected value', () => {
    const items = ['item1', 'item2', 'item3'];
    const props = {
      placeholder: 'testPlaceholder',
      // selectedValue: jest.fn().mockImplementation(() => 'value'),
      // selectedValue: jest.fn(() => 'item1'),
      selectedValue: jest.fn(() => items[0]),
      onValueChange: jest.fn(),
      items,
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<FormPicker { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });
});
