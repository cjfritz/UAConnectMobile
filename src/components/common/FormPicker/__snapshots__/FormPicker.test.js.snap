// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormPicker /> can render selected value 1`] = `
<Styled(Item)
  picker={true}
>
  <Styled(PickerNB)
    iosIcon={
      <Styled(Icon)
        color="darkred"
        name="ios-arrow-down"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    }
    onValueChange={[Function]}
    placeholder="testPlaceholder"
    renderHeader={[Function]}
    selectedValue="item1"
  >
    <Item
      label="item1"
      value="item1"
    />
    <Item
      label="item2"
      value="item2"
    />
    <Item
      label="item3"
      value="item3"
    />
  </Styled(PickerNB)>
</Styled(Item)>
`;

exports[`<FormPicker /> can render with initial state 1`] = `
<Styled(Item)
  picker={true}
>
  <Styled(PickerNB)
    iosIcon={
      <Styled(Icon)
        color="darkred"
        name="ios-arrow-down"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    }
    onValueChange={[Function]}
    placeholder="testPlaceholder"
    renderHeader={[Function]}
  >
    <Item
      label="item1"
      value="item1"
    />
    <Item
      label="item2"
      value="item2"
    />
    <Item
      label="item3"
      value="item3"
    />
  </Styled(PickerNB)>
</Styled(Item)>
`;
